src/main/java/com/google/sps/servlets/TaskServlet.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that creates new task entity and fetch saved tasks. */
+@WebServlet("/tasks")
+public class TaskServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Serve the GET request sent from home page to fetch all the tasks
+        return;
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the rewarding points from the form
+        int rewardPts = getRewardingPoints(request, "reward-input");
+        if (rewardPts == -1) {
+            response.setContentType("text/html");
+            response.getWriter().println("Please enter a valid integer in the range of 0-200");
+            return;
+        }
+
+        // Get the task detail from the form input
+        String taskDetail = "";
+        String input = request.getParameter("task-content-input");
+        if (input != null) {
+            taskDetail = input;
+        }
+
+        // If the input is nonempty and valid, set the taskDetail value to the input value
+        if (!taskDetail.equals("")) {
```

Comment:
What happens if there are no details provided?

Reply:
I actually wanted to reject the request if the task detail is not provided. So at that time, I just didn't handle the request and redirected the user back to the user profile page. Now I think maybe respond with a 403 Forbidden error can be a better idea. I will make the change in the new commit.

Reply:
[400 Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) is probably more appropriate; 403 Forbidden usually indicates a permissions error.

Reply:
Thanks! I will change it in a new commit.